packages/block-library/src/code/theme.scss

```diff
@@ -1,6 +1,10 @@
+.wp-block-code {
+	border: 1px solid #ccc;
+	border-radius: 4px;
+	box-sizing: border-box;
```

Comment:
This property keeps the border on the inside, pushing content inwards.

Reply:
I would expect this to be the default for all blocks.. Do you know why we have it in block editor but not in front end? Check the difference in a `Heading` with margin between the editor/frontend.

https://user-images.githubusercontent.com/16275880/148763840-26276997-f546-464a-8fc2-51ddf4af5bc7.mov

Reply:
Good thought, yes, it makes sense for this to apply to all blocks 🤔 

I partially recall conversations around `box-sizing` being something for the theme to apply if necessary. But if, as is the case here, there's a discrepancy between frontend and editor, that crosses a line for me. CCs at random for thoughts: @jameskoster @MaggieCabrera — could we/should we apply `box-sizing: border-box;` across all blocks? What might be a good way to do this?

Reply:
Generally many themes apply a blanket * { box-sizing: border-box; }. We do so on Blockbase + children and has generally worked well (I don't recall any inconsistencies in blocks due to the rule from the top of my head). I think we tried to add a similar general rule directly in the editor but the consensus was that it was better done on a per-block basis? I'd love it if we could aim for this, but it would probably require some testing. If I can help with this, let me know!

Reply:
Good thoughts. I wonder if maybe this is a good time to revisit a consensus. Essentially any block that opts into the border control will have a probably-not-expected effect on the frontend only, meaning if it might start to make sense at a blanket level instead. CC: @ramonjd @aaronrobertshaw since I know you worked on border support. What do you think?

Reply:
>any block that opts into the border control 

Same goes for blocks with padding/margin support.

Reply:
We'll have to be careful with things like images, and other blocks that involve wrappers. It might be strange if you set a specific width on an image, then add some padding (which gets applied to the wrapper), and find that the image then renders at a different size to the setting. I ran in to a similar issue with the border controls and the site logo block [here](https://github.com/WordPress/gutenberg/pull/37783).

Reply:
> Essentially any block that opts into the border control will have a probably-not-expected effect on the frontend only, meaning if it might start to make sense at a blanket level instead. CC: @ramonjd @aaronrobertshaw since I know you worked on border support. What do you think?

I added a bunch of blocks to a post using the Blank Canvas theme and removed the theme's box-sizing rules. Here's the frontend.

<img width="500" alt="Screen Shot 2022-01-11 at 1 02 06 pm" src="https://user-images.githubusercontent.com/6458278/148868377-cd2726a6-22ba-4a97-83e7-5fecbb37e9ea.png">

Heading and List stand out as blocks that don't take care of their own box-sizing in the absence of any theme rules.

I don't have much context on the background, but I had a poke around and, in the editor, there's a [blanket rule for box sizing](https://github.com/WordPress/gutenberg/blob/a4e3977aa2ed12d61515828d95001e531521eb51/packages/edit-post/src/style.scss#L71) based on a [reset mixin](https://github.com/WordPress/gutenberg/blob/a4e3977aa2ed12d61515828d95001e531521eb51/packages/edit-post/src/style.scss#L71).

Here's how the editor looks with and without this reset rule:

![box-sizing-blank-canvas](https://user-images.githubusercontent.com/6458278/148868183-c1b8afc3-c271-4aaf-b396-b864e8eeffd0.gif)

So it makes me think: if we have a default blanket box-sizing rule for blocks in the editor, then it does make sense that they exist on the frontend as well.

On the back of the napkin in a seedy bar after a few 🍺 I'd have written:

- add a low-specificity `box-sizing: border-box;` rule, or carry over the reset rule, for blocks on the frontend
- see if `box-sizing: border-box;` core blocks still need their specific rules
- watch things break 🍿

Reply:
Excellent thoughts, Jay, I knew there was a good reason it wasn't necessarily simple. 

And thanks so much Ramon for testing so thoroughly. Your feedback definitely suggests that either we output the blanket statement on the frontend, _or_ we remove `.edit-post-visual-editor` from the list to receive the reset, so at least editor and frontend are the same. 

The following could potentially provide a soft reset across the frontend:

```
:where(*) {
	box-sizing: border-box;
}
```

That would give it low specificity, making it trivial to override.

The thing we want to avoid, though, is what Jay is suggesting, the use case where you _want_ the border to be on the outside, but can't do it because of the rule. I suppose that's technically already the case in the editor if not the frontend, but is that a concern?

Reply:
> I suppose that's technically already the case in the editor if not the frontend

I think is the the crux: we're forcing `box-sizing: border-box;` in the editor so, if I understand the concern correctly, should we add the reset, we'd be removing the ability to "trick" the editor into showing the border on the outside on the frontend.  (?)

Another concern might be that the editor styles rules are rather specific:

<img width="426" alt="Screen Shot 2022-01-12 at 2 21 37 pm" src="https://user-images.githubusercontent.com/6458278/149058204-2b76da75-cbf0-4ee1-9b89-391f5cdc677d.png">

Rather than a blanket rule across all elements, would we need to target blocks in a post wrapper element instead?

Anyway, I've thrown up a [draft PR](https://github.com/WordPress/gutenberg/pull/37902) if we want to move the discussion over to there. 

Thanks for the CSS suggestion @jasmussen !